### ISSUE TITLE
[bug]boost 1.86.0 package is missing process::v2 libs

### ISSUE THREAD

**Issue by bmwrightson** [conan-io/conan#17065]
### Describe the bug

As of boost 1.86.0 process::v2 is no longer header only (see https://www.boost.org/users/history/version_1_86_0.html) but the libs for it are not packaged in the conan recipe, leading to unresolved externals.

### How to reproduce it

```
#include <boost/process/v2/environment.hpp>

int main()
{
    auto path = boost::process::v2::environment::get("PATH");
}

```

**Comment by uilianries** [comment]
@bmwrightson Yes, indeed boost process is now a library and not a header-only in Boost 1.86.0. The PR #25388 is working on it. Thank you for reporting.

**Comment by memsharded** [comment]
I am moving this to conan-center-index repo then, not a Conan bug, but a recipe issue
